src/posts/lucky_1.2.0.cr

```diff
@@ -0,0 +1,237 @@
+    CardQuery.new.title.reverse.eq("tacocat")
+
+    # Find all purchases where the amount rounded up is greater than 5
+    PurchaseQuery.new.amount.ceil.gt(5)
+
+    # Turn a negative number in to positive, and use in your query
+    PointQuery.new.x_position.abs.lte(1)
+    ```
+
+    To add to these great methods, we've also added `JSON::Any` critera methods. Querying jsonb fields
+    can be a bit tricky, and Avram now allows you to avoid a few unsafe raw WHERE queries by using type-safe ones.
+
+    ```crystal
+    # where the preferences jsonb has the key 'theme'
+    UserQuery.new.preferences.has_key("theme")
+
+    # where the preferences jsonb has the key 'theme' or the key 'color'
+    UserQuery.new.preferences.has_any_keys(["theme", "color"])
+
+    # where the preferences jsonb has the key 'theme' and the key 'color'
+    UserQuery.new.preferences.has_all_keys(["theme", "color"])
+
+    # where the preferences jsonb contains the JSON {"theme": "dark"}
+    UserQuery.new.preferences.includes({theme: "dark"})
+
+    # where the preferences jsonb is found in this json document
+    UserQuery.new.preferences.in(large_json_document)
+    ```
+
+    Finally, `String` also received a few more Criteria related to Full-Text searching.
+
+    Convert a `String` column `to_tsquery` or `to_tsvector` by using those methods names in
+    conjunction with the new `match` method.
+
+    ```crystal
+    # SELECT * FROM profiles WHERE TO_TSVECTOR(description) @@ TO_TSQUERY('Lucky')
+    ProfileQuery.new.description.to_tsvector.match("Lucky")
+    ```
+
+    ### New migration methods
+
+    Migrations received a few updates. You can now run your `create` and `alter` conditionally
+    with `if_not_exists` and `if_exists` respectively.
+
+    ```crystal
+    # Only create the table if it does not already exist
+    create table_for(Server), if_not_exists: true do
+      add hostname : String
+    end
+    ```
+
+    You can now ignore the index created on `add_belongs_to` by specifying `index: false`.
+
+    ```crystal
+    alter table_for(Reaction), if_exists: true do
+      add_belongs_to user : User?, index: false
+      add_belongs_to friend : User?, index: false
+    end
+    ```
+
+    This will allow you to create a custom unique index utilizing multiple columns instead of separate indexes on each column.
+
+    We've also added the `create_sequence` and `drop_sequence` helpers.
+
+    ```crystal
+    def migrate
+      create_sequence table_for(Account), :number
+    end
+
+    def rollback
+      drop_sequence table_for(Account), :number
+    end
+    ```
+
+    ### The IGNORE constant
+
+    In Avram, the attributes defined on operations are a union of the type you specified, and `Avram::Nothing`.
+    The `Avram::Nothing` type is used as a way to differentiate between you intending to update a column to `nil`
+    by passing in a `nil` value versus a `nil` value existing in params but with no intention of updating the column.
```

Comment:
This was really confusing and I did my best. Is this correct? Is there something that I'm leaving out that's important here? I can't find any existing docs to check my knowledge here.

```suggestion
      The `Avram::Nothing` type was used when updating a model to specify that a column should be set to `NULL`. This is instead of ignoring that column in the update statement entirely.
```

It may also help to show a before and after.

Reply:
That's not quite correct. The `Avram::Nothing` type didn't go away. `IGNORE` is just defined as `IGNORE = Avram::Nothing.new`. It's used when you are passing in `nil` but you *don't* want to set the value to `NULL`. The real benefit here is just that instead of having 1000 instances of `Avram::Nothing`, you now have 1. Does that clarify better?

Reply:
Reworded this. It's a bit complicated to explain, but let me know if that reads better.